src/app/static/src/app/store/load/actions.ts

```diff
@@ -70,6 +71,7 @@ export const actions: ActionTree<LoadState, RootState> & LoadActions = {
             await (dispatch("projects/createProject", projectName, {root: true}));
             savedState.projects.currentProject = rootState.projects.currentProject;
             savedState.projects.currentVersion = rootState.projects.currentVersion;
+            savedState.stepper.steps = initialStepperState().steps.filter(step => step.number <= 2);
```

Comment:
This is close but not quite what's needed. We don't want to blitz all of the existing stepper state - this will remove all but the first two steps. We just to replace the text keys with those from `initialStepperState()`.
I think we can also move this outside that `if` clause - this should work when a guest is loading from a saved file, as well as when a logged in user loads from a project.  
UPDATE: Having tested this suggested change, it does work for loading a saved file but appears to have no effect when loading a previously saved project (we still see the untranslated new text keys) so there must be some difference in the load logic which I'm missing
```
const latestKeys = initialStepperState();
savedState.stepper.steps.forEach((step: StepDescription) => {
  step.textKey = latestKeys.steps.find(latestStep => latestStep.number == step.number)!.textKey;
});

if (!rootGetters.isGuest) {
    await (dispatch("projects/createProject", projectName, {root: true}));
    savedState.projects.currentProject = rootState.projects.currentProject;
    savedState.projects.currentVersion = rootState.projects.currentVersion;
}
```

Reply:
Furhter update: OK, I think I figured it out, it's because load from project goes through a different code path, and doesn't hit load/setFiles action. So we can take the change from there, and put it into a slightly later action which is shared by both methods. If I add the line `
savedState.stepper.steps = initialStepperState().steps; //ensure latest step translation keys
`
at the start of load/getFilesAndLoad, then the load with new step names works for me, for both loading files as guest and loading projects.